internal/cli/version.go

```diff
@@ -0,0 +1,185 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/newrelic/newrelic-cli/internal/output"
+	"github.com/newrelic/newrelic-cli/internal/utils"
+)
+
+var (
+	// Represents the currently installed/compiled version of the CLI
+	version string
+
+	// Represents the latest release version per the GitHub repository's latest release
+	latestVersion string
+)
+
+const installCLISnippetLinux = `curl -Ls https://raw.githubusercontent.com/newrelic/newrelic-cli/master/scripts/install.sh | bash && sudo newrelic install`
+const installCLISnippetWindows = `[Net.ServicePointManager]::SecurityProtocol = 'tls12, tls'; (New-Object System.Net.WebClient).DownloadFile("https://github.com/newrelic/newrelic-cli/releases/latest/download/NewRelicCLIInstaller.msi", "$env:TEMP\NewRelicCLIInstaller.msi"); msiexec.exe /qn /i $env:TEMP\NewRelicCLIInstaller.msi | Out-Null; & 'C:\Program Files\New Relic\New Relic CLI\newrelic.exe' install`
+
+// NewRelicCLILatestReleaseURL is the URL used to fetch the latest release data.
+const NewRelicCLILatestReleaseURL string = "https://api.github.com/repos/newrelic/newrelic-cli/releases/latest"
+
+// NewRelicCLILatestReleaseURL is the URL used to fetch the latest release data.
+const PrereleaseEnvironmentMsgFormat string = `
+  It appears you're in a development environment using prerelease version %s.
+  To upgrade the New Relic CLI, you must be using non-prerelease version.
+`
+
+// UpdateVersionMsgFormat is the message displayed to a user when an older
+// version of the CLI is installed.
+const UpdateVersionMsgFormat string = `
+  We need to update your New Relic CLI version to continue.
+
+    Installed version: %s
+    Latest version:    %s
+
+  To update your CLI and continue this installation, run this command:
+
+    %s
+`
+
+// Version returns the version of the CLI that's currently installed.
+func Version() string {
+	if version == "" {
+		version = os.Getenv("NEW_RELIC_CLI_VERSION")
```

Comment:
Why is this tied to an environment variable?  If it's for testing purposes, can't we just update the value of the `version` field from within the tests?

Reply:
It was due to another test that utilizes an environment variable. Here is [the test](https://github.com/newrelic/newrelic-cli/blob/main/internal/install/recipe_installer_test.go#L904) that uses it, and [here](https://github.com/newrelic/newrelic-cli/blob/c9fe182cc5b8893ba5187f7e968d36aee4b25206/internal/install/execution/install_status.go#L386-L393) is where it was implemented.